modules/ROOT/pages/cloudhub-impaired-worker.adoc

```diff
@@ -1,17 +1,48 @@
 
-When a worker is impaired, CloudHub automatically attempts to restart the worker and relaunch the application. If a retry is successful, the underlying worker and the Mule application are recovered without requiring any further action.
+CloudHub continuously performs health checks on the underlying virtual machines where Mule applications are running. 
+When a worker is impaired, CloudHub automatically attempts to restart the worker and relaunch applications.
 
-[NOTE]
+A worker is impaired when there is a hardware or software failure that negatively impacts the running application deployed on the worker.
+Hardware and software failures include network unavailability, memory issues, server hardware failures, and file corruption.
+
+When CloudHub restarts the impaired worker:
+
+* If a retry is successful, the underlying worker and app restart without any further action.
++
 It might take up to five minutes for a worker to restart.
+* If the application fails to start after five attempts, the status of the application is *Failed* or *Unrecoverable Runtime Error*.
++
+In this case, you must manually restart the application.
 
-If the application fails to start after five attempts, the status of the application is *Failed* or *Unrecoverable Runtime Error*. In this case, you must manually redeploy and restart the application.
+The logs indicate when a worker is restarted automatically and any failures.
+
+[NOTE]
+The automatic worker-restart functionality is different from the automatic application-restart functionality, which is configurable using the *Automatically restart application when not responding* option on the *Settings* page for the app. 
+CloudHub automatically attempts to restart impaired workers, regardless of the app setting. 
 
 == Scheduled Maintenance Activities
 
-CloudHub checks for scheduled maintenance activities on the worker instances running the Mule applications. The following maintenance activities can cause a potential downtime for the running applications: reboot, stop, start, or retirement of the instances. To avoid downtime, Cloudhub checks for these scheduled events and sends you a notification in advance to restart your application with zero downtime.
+CloudHub checks for scheduled maintenance activities for worker instances that cause potential downtime for the running applications: reboot, stop, start, or retirement of the instances.
+
+A notification appears in the top right of the *Settings* page for the application and on the *Applications* page, notifying you of upcoming maintenance activities:
+
+.The arrow shows the notification icon in the *Settings* page.
+image::cloudhub-notification-icon.png[Notification icon in the *Settings* page]
+
+Click the icon to display the notification, which includes the date that CloudHub will automatically restart the worker.
+You can restart your apps manually or wait for CloudHub to restart them automatically. 
```

Comment:
You can restart your apps manually or wait for CloudHub to restart them automatically. A reminder notification will appear 1 week before the restart. 

If the platform performs the restart on your behalf, you will see the following message in the application log:
"Your application was running on an impaired worker instance. To avoid any downtime, the platform has restarted the application."
You will also see a notification indicating that the restart was done.

Reply:
@hannanelson